### DISCUSSION TITLE
add `magit-process-kill` command in magit-process buffers

### DISCUSSION THREAD

**Discussion by sellout** [magit/magit#4921] [Ideas]
I’ve had occasions where something has messed up some git command. Most recently, I did something that left me without an editor buffer, but `git … commit` is sitting there with “Waiting for your editor to close the file…”

I would love a key-bound `magit-process-kill` available in magit-process buffers. Even better if it cleans up the lock file, too.

**Comment by tarsius** [comment]
I've added a binding in b0ac2dbd674a021d24f2a70a04c0627d2d21170e.

> Even better if it cleans up the lock file, too.

Usually `git` should leave no lock file behind anyway. So we can probably get away without that. We would have to somehow determine that the look file was left behind by the process we just killed and if I remember correctly, that isn't actually possible.
